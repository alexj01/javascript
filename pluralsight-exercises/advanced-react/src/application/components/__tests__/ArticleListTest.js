import React from 'react';
import ArticleList from '../ArticleList';
// import renderer from 'react-test-renderer';
import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({ adapter: new Adapter() });
const { shallow } = enzyme;

describe('ArticleList', () => {

    const testProps = {
        articles: {
            a: { id: 'a', date: 'Sun Jul 10 2016', title: 'hello', body: '1234helloagain'},
            b: { id: 'b', date: 'Sun Jul 10 2016', title: 'hello', body: '1234helloagain'},
        },
        store: {
            lookupAuthor: jest.fn(()=>({}))
        }
    }

    it('renders correctly', () => {
        const wrapper = shallow(
            <ArticleList
                {...testProps}
            />
        );

        // console.log(wrapper.getElement())
        expect(wrapper.getElement().props.children).toMatchSnapshot();
        expect(wrapper.getElement().props.children.length).toBe(2);
        expect(wrapper.find('ArticleContainer').length).toBe(2);
        // expect(element).toBe(true);
    });
});