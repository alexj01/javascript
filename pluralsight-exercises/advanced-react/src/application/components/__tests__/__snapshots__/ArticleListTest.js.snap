// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
Array [
  <ArticleContainer
    article={
      Object {
        "body": "1234helloagain",
        "date": "Sun Jul 10 2016",
        "id": "a",
        "title": "hello",
      }
    }
  />,
  <ArticleContainer
    article={
      Object {
        "body": "1234helloagain",
        "date": "Sun Jul 10 2016",
        "id": "b",
        "title": "hello",
      }
    }
  />,
]
`;
